smithy-swift-codegen/src/main/kotlin/software/amazon/smithy/swift/codegen/integration/serde/readwrite/ReadingClosureUtils.kt

```diff
@@ -10,34 +10,35 @@ import software.amazon.smithy.model.traits.XmlFlattenedTrait
 import software.amazon.smithy.swift.codegen.SwiftWriter
 import software.amazon.smithy.swift.codegen.integration.ProtocolGenerator
 import software.amazon.smithy.swift.codegen.integration.serde.json.TimestampUtils
-import software.amazon.smithy.swift.codegen.integration.serde.xml.NodeInfoUtils
 import software.amazon.smithy.swift.codegen.model.getTrait
 import software.amazon.smithy.swift.codegen.model.hasTrait
 
 class ReadingClosureUtils(
     val ctx: ProtocolGenerator.GenerationContext,
     val writer: SwiftWriter
 ) {
-    private val nodeInfoUtils = NodeInfoUtils(ctx, writer)
+    private val nodeInfoUtils = NodeInfoUtils(ctx, writer, ctx.service.responseWireProtocol)
 
-    fun readingClosure(member: MemberShape): String {
+    fun readingClosure(member: MemberShape, isSparse: Boolean = false): String {
         val target = ctx.model.expectShape(member.target)
         val memberTimestampFormatTrait = member.getTrait<TimestampFormatTrait>()
-        return readingClosure(target, memberTimestampFormatTrait)
+        val baseClosure = readingClosure(target, memberTimestampFormatTrait)
+        if (isSparse) {
+            return writer.format("optionalFormOf(readingClosure: \$L)", baseClosure)
+        } else {
+            return baseClosure
+        }
```

Comment:
Creates a special / modified reading closure for use in sparse collections.

Also some changes to closure forms below.